src/shillelagh/backends/apsw/vt.py

```diff
@@ -547,8 +582,8 @@ def Filter(  # pylint: disable=too-many-locals
             kwargs["limit"] = limit
         if self.adapter.supports_offset:
             kwargs["offset"] = offset
-        if self.adapter.supports_requested_columns:
-            kwargs["requested_columns"] = self.requested_columns
+        if best_index_object_available() and self.adapter.supports_requested_columns:
```

Comment:
as previously mentioned, you can drop the check here and just infer based on the presence of a `key` in the dict.

The version based check `best_index_object_available()` might be slow in the query path.